src/common/utils/get_resolved_document_by_id.py

```diff
@@ -9,8 +9,8 @@ def resolve_reference_list(x: list, document_repository: DataSource, depth: int
     if isinstance(x[0], list):  # Call recursively for nested lists
         resolved = [resolve_reference_list(item, document_repository) for item in x]
     for value in x:
-        if isinstance(value, dict) and value.get("_id"):  # It's a reference!
-            resolved.append(get_complete_sys_document(value["_id"], document_repository, depth, depth_count))
+        if isinstance(value, dict) and value.get("ref"):  # It's a reference!
```

Comment:
Should check for type, instead of asuming that any entity with a "ref" key is a reference